packages/accordion/components/AccordionItem.tsx

```diff
@@ -22,16 +21,18 @@ export interface AccordionItemProps {
 const AccordionItem = ({
   children,
   "data-cy": dataCy = "accordionItem",
-  id = nextId("accordionItem-"),
+  id,
   className
 }: AccordionItemProps) => {
+  const generatedId = `accordionItem-${React.useId()}`;
+  const accordionItemId = id || generatedId;
```

Comment:
No big deal, but this way we always generate an ID even though we pass in an `id`.
It would be a bit more performant if `generateId` is a function which is only executed when we call it - but I guess that's not easily possible with a hook.
We could try to build a custom hook that does that for us.
Or a simple solution may be something like:

```
  const accordionItemId = id || `accordionItem-${React.useId()}`;
```

But anyway, I don't think generating an id is that much of a computing overhead, so decide for yourself if you want to improve that or not.

Reply:
I wanted to do that method with one line but it will cause this `React Hook "React.useId" is called conditionally. React Hooks must be called in the exact same order in every component render.` so that's why i've broken it out into 2 lines. I will consider the computing overhead a bit more today before merging, glad that it's not much of one.